core/iwasm/libraries/lib-socket/test/nslookup.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <string.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#endif
+#include <netdb.h>
+
+void
+testNsLookup(int af)
+{
+
+    struct addrinfo *res;
+    int count = 0;
+    struct addrinfo hints;
+    char *url = "google-public-dns-a.google.com";
+
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = af;
+    hints.ai_socktype = SOCK_STREAM;
+    int ret = getaddrinfo(url, 0, &hints, &res);
+    assert(ret == 0);
+    if (ret == 0) {
```

Comment:
we already have the line above to check for the return code, that should be enough since this is a test (i.e. not meant to be run without the asserts)